R/derive_param_diam_to_sev.R

```diff
@@ -0,0 +1,226 @@
+#'  records by passing the appropriate `filter_diam`, `filter_faobj`and severity
+#'  grade limits.
+#'
+#' @return
+#' The Input data with the new severity records for Redness and swelling which
+#' is specified in `filter_faobj` and AVAL, AVALC will be derived and fatestcd,
+#'  fatest will be changed as per the values
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(admiral)
+#' library(admiraldev)
+#' library(tibble)
+#' Input <- tribble(
+#'   ~USUBJID, ~FAOBJ, ~AVAL, ~AVALC, ~ATPTREF,
+#'   "XYZ1001", "REDNESS", 7.5, 7.5, "VACCINATION 1",
+#'   "XYZ1001", "REDNESS", 3.5, 3.5, "VACCINATION 1",
+#'   "XYZ1001", "REDNESS", 2, 2, "VACCINATION 1",
+#'   "XYZ1001", "REDNESS", 1.8, 1.8, "VACCINATION 1",
+#'   "XYZ1001", "REDNESS", 1.4, 1.4, "VACCINATION 1",
+#'   "XYZ1002", "REDNESS", 11.1, 11.1, "VACCINATION 2",
+#'   "XYZ1002", "REDNESS", 7.4, 7.4, "VACCINATION 2",
+#'   "XYZ1002", "REDNESS", 6, 6, "VACCINATION 2",
+#'   "XYZ1002", "REDNESS", 2.1, 2.1, "VACCINATION 2",
+#'   "XYZ1002", "REDNESS", 1.1, 1.1, "VACCINATION 2",
+#'   "XYZ1001", "SWELLING", 5.5, 5.5, "VACCINATION 1",
+#'   "XYZ1001", "SWELLING", 2.5, 2.5, "VACCINATION 1",
+#'   "XYZ1001", "SWELLING", 2, 2, "VACCINATION 1",
+#'   "XYZ1001", "SWELLING", 1.8, 1.8, "VACCINATION 1",
+#'   "XYZ1001", "SWELLING", 1.4, 1.4, "VACCINATION 1",
+#'   "XYZ1002", "SWELLING", 10.1, 10.1, "VACCINATION 2",
+#'   "XYZ1002", "SWELLING", 7.1, 7.1, "VACCINATION 2",
+#'   "XYZ1002", "SWELLING", 5, 5, "VACCINATION 2",
+#'   "XYZ1002", "SWELLING", 1.8, 1.8, "VACCINATION 2",
+#'   "XYZ1002", "SWELLING", 1.4, 1.4, "VACCINATION 2"
+#' ) %>% mutate(
+#'   AVALC = as.character(AVALC),
+#'   FATEST = "Diameter",
+#'   FATESTCD = "DIAMETER"
+#' )
+#'
+#' output <- derive_param_diam_to_sev(
+#'   dataset = Input,
+#'   filter_faobj = c("REDENSS", "SWELLING"),
+#'   filter_diam = "DIAMETER",
+#'   testcd_sev = "SEV",
+#'   test_sev = "Severity"
+#' )
+#' @keywords der_adxx
+#' @family der_adxx
+derive_param_diam_to_sev <- function(dataset = NULL,
+                                     filter_diam = "DIAMETER",
+                                     filter_faobj = c("REDNESS", "SWELLING"),
+                                     testcd_sev = "SEV",
+                                     test_sev = "Severity/Intensity",
+                                     none = c(0, 2),
+                                     mild = c(2, 5),
+                                     mod = c(5, 10),
+                                     sev = 10) {
+  #-------------------------------------------------------------------------------
+  # assertion checks
+  #-----------------------------------------------------------------------------
+  admiraldev::assert_data_frame(dataset,
+    required_vars = vars(
+      USUBJID, AVAL, AVALC,
+      FAOBJ, FATEST, FATESTCD
+    )
+  )
+  admiraldev::assert_numeric_vector(arg = c(none, mild, mod, sev), optional = FALSE)
+  admiraldev::assert_character_vector(
```

Comment:
```suggestion
 :assert_data_frame(dataset,
    required_vars = vars(
      USUBJID, AVAL, AVALC,
      FAOBJ, FATEST, FATESTCD
    )
  )
  assert_numeric_vector(arg = c(none, mild, mod, sev), optional = FALSE)
  assert_character_vector(
```

Reply:
CAn you do this in your PR for admiralvaccines like we do in admiral?  Then the `admiraldev::` can be dropped going forward
![image](https://user-images.githubusercontent.com/10111024/218784416-f2bab215-25ed-4909-b178-24450ee5490d.png)